param/parameterized.py

```diff
@@ -1684,14 +1693,12 @@ def __db_print(self_,level,msg,*args,**kw):
         See python's logging module for details.
         """
         self_or_cls = self_.self_or_cls
-        if get_logger().isEnabledFor(level):
+        if get_logger(name=self_or_cls.name).isEnabledFor(level):
 
             if dbprint_prefix and callable(dbprint_prefix):
-                prefix=dbprint_prefix() # pylint: disable-msg=E1102
-            else:
-                prefix=""
+                msg = dbprint_prefix() + ": " + msg  # pylint: disable-msg=E1102
 
-            get_logger().log(level, '%s%s: '+msg, prefix, self_or_cls.name, *args, **kw)
```

Comment:
Is this going to change the formatting of log messages?

Reply:
Yes, a warning now looks like this:

>WARNING:param.Curve01322: Some warning

It used to look like this:

>WARNING:root:Curve01315: Some warning

I'd prefer it if it was something like:

>WARNING:param:Curve01315: Some warning

but the ``.`` separator is the canonical way to define logger descendants.

Reply:
Ok.  Sounds fine, and I agree with your preference, but it's not too bad.